src/lab12.py

```diff
@@ -1361,10 +1361,21 @@ def handle_quit(self):
 
 if __name__ == "__main__":
     import sys
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Chapter 12 code')
+    parser.add_argument("--url", default=2, type=str, required=True,
+        help="URL to load")
+    parser.add_argument('--single_threaded', type=bool, default=False,
+        help='Whether to run the browser without a browser thread')
```

Comment:
I think instead of `type=bool` you want `action="store_true"`. Argparse is pretty annoying.